Migrate to connexion v3

sudiptob2 opened the pull request:
This PR is created based on https://github.com/apache/airflow/pull/36052

## Todo

- [x]  Taks 1 - Refactor [get_api_endpoints()](https://github.com/apache/airflow/blob/0b9e2a78ccf82c12bb0c2fa052958188eb93bb62/airflow/providers/fab/auth_manager/fab_auth_manager.py#L152) as it can not return `blueprint` in connexion v3 
- [x]  Taks 2 - Find a way to replace the Legacy environ_overrides method from the old version of Connexion to fix the problem in tests.
- [x]  Task 3 - Fix unit tests that use app

##  Taks 1 - Refactor get_api_endpoints()
### Problem Definiton

Ref: [Github Pull Request #36052 VladaZakharova commented on Jan 18](https://github.com/apache/airflow/pull/36052#issuecomment-1898612794)

In the `init_api_auth_provider` method, we update the base path as follows:
```
def init_api_auth_provider(connexion_app: connexion.FlaskApp):
    """Initialize the API offered by the auth manager."""
    auth_mgr = get_auth_manager()
    blueprint = auth_mgr.get_api_endpoints(connexion_app)
    if blueprint:
        base_paths.append(blueprint.url_prefix if blueprint.url_prefix else "")
        flask_app = connexion_app.app
        flask_app.extensions["csrf"].exempt(blueprint)

```
However, the `blueprint` object obtained from `auth_mgr.get_api_endpoints(connexion_app)` will always be `None` if we are using **ConnexionV3**.

### Proposed solution
Ref [vincbeck commented on Jan 18](https://github.com/apache/airflow/pull/36052#issuecomment-1898786641)

- Rename `get_api_endpoints` to `set_api_endpoints`. The return type should be updated to `None`. Documentation should be updated as well to something like "Set API endpoint(s) definition for the auth manager.". This is a breaking change but nobody uses this interface yet, so it is a good time to do it.
- ~~This piece of code `flask_app.extensions["csrf"].exempt(blueprint)` should be moved in the `set_api_endpoints` method using `appbuilder.app.extensions["csrf"].exempt(api.blueprint)`~~ Ref: https://github.com/apache/airflow/pull/37638#issuecomment-1962060087

### How to test
- **Run client tests** `python ./clients/python/test_python_client.py`

### Subtasks

- [ ]  Hanlde logic for `base_paths.append(blueprint.url_prefix if blueprint.url_prefix else "")`
- [x]  Make sure the base path for the `CSRF` exemption is correct. Ref: https://github.com/apache/airflow/pull/37638#discussion_r1509398492
- [x] Handle `swagger-ui` installation, Ref: https://github.com/apache/airflow/pull/37638#discussion_r1502453187
- [ ] Solve `favicon.ico` problem in the swagger UI. 

## Task 2 - Replace envrion_overrides argument
### Problem Definition 
Ref: [Github Pull Request #36052 commented on Feb 6](https://github.com/apache/airflow/pull/36052#:~:text=for%20available%20options)

>Find a way to replace legacy environ_overrides method from old version of Connexion to fix the problem in tests
TypeError: get() got an unexpected keyword argument 'environ_overrides'. Contact @RobbeSneyders for some help. Currently we didn't find anything in their documentation regarding new way how to override env variables.

### Solution
- Use headers to send `{'REMOTE_USER:"user"}` instead of using environ_overirdes argument inside testclient.method. Accordinly update the authentication part in  `tests/test_utils/remote_user_api_auth_backend.py` to `user_id = request.headers.get("REMOTE_USER")`

Commit:
Update methods to use Connexion v3, Ginucorn command and encoding

Commit:
Update methods to use Connexion v3, Ginucorn command and encoding

Commit:
Fix static checks

MaksYermak committed:
Update setup.cfg and encoding

MaksYermak committed:
Update migrations files for

MaksYermak committed:
Update configuration for tests

MaksYermak committed:
Fix problem with static checks

sudiptob2 committed:
fix: add missing import which was removed while rebasing, add connexion dependencies in pyproject.toml.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: fix static check.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
changed set with get

Satoshi-Sh committed:
Refactored the code since the functions return None. Replaced 'get' with 'set'

sudiptob2 committed:
feat: implement brefore_request to handle CSRF exemption logic.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Taragolis commented on pyproject.toml:
```diff
@@ -83,7 +83,7 @@ dependencies = [
     # Either revert that change or find another way, preferably without using connexion internals.
     # This limit can be removed after https://github.com/apache/airflow/issues/35234 is fixed
-    "connexion[flask]>=2.10.0,<3.0",
+    "connexion[flask,uvicorn,swagger-ui]>=3.0",
```
Right now we [install swagger via npm](https://github.com/apache/airflow/pull/28788) package [`swagger-ui-dist`](https://www.npmjs.com/package/swagger-ui-dist) and attach it to the Connexion, so we should do one of the action:
- Remove `swagger-ui-dist` and and use [`swagger-ui-bundle`](https://github.com/spec-first/swagger_ui_bundle) instead. In this case dependecy bot do not show should we update swagger version or not, see: https://github.com/apache/airflow/pull/28824
- Do not use [swagger-ui-bundle](https://github.com/spec-first/swagger_ui_bundle) in this case we need to provide path to the swagger ui into the Connexion options

Satoshi-Sh replied:
Thanks for the feedback. 

I believe this is related to "Properly setup bundled swagger-ui in Connexion #36975". 
When I opened the swagger page on this branch, I got the response `{"type": "about:blank", "title": "Not Found", "detail": "The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.", "status": 404}`

We will make task 4 about this and will work on it later.

Satoshi-Sh replied:
@Taragolis 

On the connexion v3 branch, the `/redoc` and `/api/v1/ui` didn't work. I modified the code to let them work. I also removed swagger-ui-dist.

Can you have a look at [this PR](https://github.com/sudiptob2/airflow/pull/8)  Any feedback is appreciated.

Taragolis replied:
It doesn't work with current implementation with `swagger-ui-dist` because is should be use `swagger_ui_template_dir` not a `swagger_ui_path`: https://connexion.readthedocs.io/en/stable/swagger_ui.html#connexion.options.SwaggerUIOptions

potiuk replied:
Hey @Satoshi-Sh -> I looked at the history of swaggers, and I think we should attempt to update the swagger-ui-dist instead (sorry :(  - I hope setting up the things in https://github.com/sudiptob2/airflow/pull/8 and the hint from @Taragolis abobve - will help configuring the dist one. 

The thing is that the one that we can install from PyPI is already pretty old - and it's only a question of time when some vulnerabilities will be found in it. If you look here - https://pypi.org/project/swagger-ui-bundle/1.1.0/#files the one from November bundles the javascript for swagger `4.15.5` which has been released a year ago https://www.npmjs.com/package/swagger-ui-dist/v/4.15.5 where we are currently at https://www.npmjs.com/package/swagger-ui-dist/v/5.11.8 released 7 days ago. So if we could attempt to use the sdist one - that is way better. However it might turn  out that there will be a compatibility problem with the connexion and 5.* line of swagger - @RobbeSneyders - maybe you can comment on that - do you foresee any problem with attempting to use swager-ui-dist  5.* version with Connexion 3 ?

Satoshi-Sh replied:
Thanks for the pointer Jarek. I will have a look at it.

Satoshi-Sh replied:
I made another [PR](https://github.com/sudiptob2/airflow/pull/13)

It's working fine other than the Redoc page. It cannot load favicon.ico. I need to investigate about it. 

I upgraded the swagger-ui-dist to 5.11.8. It seems working fine as before.

potiuk replied:
Sounds great. I think you can even merge that one into @sudiptob2 PR and we can solve the `favicon.ico` later.

vincbeck commented on airflow/providers/fab/auth_manager/fab_auth_manager.py:
```diff
@@ -149,19 +150,24 @@ def get_cli_commands() -> list[CLICommand]:
-            validator_map={"body": _CustomErrorRequestBodyValidator},
-        ).blueprint
+        )
+        return None
```
Not necessary

```suggestion
```

vincbeck commented on airflow/www/app.py:
```diff
@@ -70,7 +72,27 @@
+    def before_request():
+        """Exempts the view function associated with '/api/v1' requests from CSRF protection."""
+        if request.path.startswith("/api/v1"):  # TODO: make sure this path is correct
+            view_function = flask_app.view_functions.get(request.endpoint)
```
```suggestion
            view_function = connexion_app.app.view_functions.get(request.endpoint)
```

vincbeck commented on airflow/www/extensions/init_views.py:
```diff
@@ -323,11 +301,7 @@ def init_api_experimental(app):
 
 
-def init_api_auth_provider(app):
+def init_api_auth_provider(connexion_app: connexion.FlaskApp):
```
I know it might not be related to this change but `auth_provider` no longer make sense, we should update it as well

```suggestion
def init_api_auth_manager(connexion_app: connexion.FlaskApp):
```

vincbeck reviewed (commented):
Few minor comments but overall LGTM!! Very good work!

sudiptob2 committed:
test: adapt broken unit test due to connexion_app

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

potiuk commented on airflow/www/extensions/init_views.py:
```diff
@@ -323,11 +301,7 @@ def init_api_experimental(app):
-        base_paths.append(blueprint.url_prefix)
-        app.register_blueprint(blueprint)
-        app.extensions["csrf"].exempt(blueprint)
+    auth_mgr.set_api_endpoints(connexion_app)
```
We should make sure to review the exemption removals.

cc: to myself.,

Taragolis commented on airflow/cli/commands/webserver_command.py:
```diff
@@ -383,7 +383,7 @@ def webserver(args):
             str(num_workers),
             "--worker-class",
-            str(args.workerclass),
+            "uvicorn.workers.UvicornWorker",
```
Is this the only one supported worker after switch to the Connexion 3? If so we still need to get this value from the airflow configuration [`[webserver] worker_class`](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#worker-class).

It code be changed to default, and if user provide any other values, then replace it to the  `uvicorn.workers.UvicornWorker` with future warning, some current implementation could be found into the [`airflow.configuration.AirflowConfigParser.validate`](https://github.com/apache/airflow/blob/f36e1c75dc47dd29ad89d66ce357e333174cd29f/airflow/configuration.py#L717)
 
In additional newsfragment should be created with [description of significant changes](https://github.com/apache/airflow/tree/main/newsfragments)

potiuk replied:
I believe this is just a temporary change to test it, but yes - we shoudl keep the option to be able to use original workers (here and in the internal_api). So we should leave that conversation open till then.

Taragolis replied:
>[Running the application](https://connexion.readthedocs.io/en/latest/v3.html#running-the-application)
There have been 2 changes related to running the application:
>
>You now MUST run the Connexion application instead of the underlying Flask application.
>You should use an ASGI server instead of a WSGI server.

So I guess there is no way to run Connexion 3 with default gunicorn workers because it is implements WSGI and `uvicorn.workers.UvicornWorker` adds ASGI support or it works as middleware, not sure. But maybe I miss something

potiuk replied:
Well. Cool then we have to document that we change to uvicorn. We have no othe way I think and IMHO gunicorn worker configuration is really not something that we promised - so yes. significant description and changing to uvicorn should be documented and we should likely - when we get closer to merging that one, review the configuration and documentation

Taragolis replied:
I guess so, that it not something really braking/important. I could say even more we only check with `sync`, all other potentially doesn't work well, or we have to patch them (gevent). I think most of the user even doesn't noticed that

potiuk replied:
Yep

Taragolis commented on airflow/cli/commands/internal_api_command.py:
```diff
@@ -101,7 +101,7 @@ def internal_api(args):
             str(num_workers),
             "--worker-class",
-            str(args.workerclass),
+            "uvicorn.workers.UvicornWorker",
```
Same here

potiuk commented on airflow/cli/commands/internal_api_command.py:
```diff
@@ -73,8 +73,8 @@ def internal_api(args):
-            debug=True,  # nosec
-            use_reloader=not app.config["TESTING"],
+            log_level="debug",
+            # reload=not app.app.config["TESTING"],
```
Just a reminder to restore that one later.

potiuk commented on airflow/www/app.py:
```diff
@@ -70,7 +72,27 @@
+    @connexion_app.app.before_request
+    def before_request():
+        """Exempts the view function associated with '/api/v1' requests from CSRF protection."""
+        if request.path.startswith("/api/v1"):  # TODO: make sure this path is correct
```
Answering the question about the paths here - this execamption shoult be done for all the API endpoints - so basicaly anything that is stored in ` init_views.base_paths` -> all the API  endpoints will register itself there - for example:

```python
    base_path = "/api/v1"
    base_paths.append(base_path)
```

And and - we should be able to add exemption when path starts with any of the api urls registered there.

Satoshi-Sh committed:
handle swagger ui installation.

Satoshi-Sh committed:
Adapt unittest with environ override.

sudiptob2 committed:
fix: fixing test_auth of connexion api.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: fix react www test

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt few test of www module.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

potiuk committed:
Add asset compilation when testing openapi client

potiuk committed:
Add Pytest fixture to create directory that starlette needs

sudiptob2 committed:
fix: fix failing static check.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
Fixed StaleDataError by adding session.refresh(user)

Satoshi-Sh committed:
Added '/auth/fab/v1' to the base_paths to avoid coroutine not callable errors

Satoshi-Sh committed:
Modified assert 'title' and 'type' accodringly.

sudiptob2 committed:
fix: test_should_respond_200_with_anonymous_user fixed.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: unit tests of experimental/test_dag_runs_endpoint.py.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt unit test to check for redirection.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
fix: Added 'init_jinja_globals' to minimal app. Updated client_with_login to use path instead of status code.

sudiptob2 committed:
fix: adapt unit test in www/test_views

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt unit test with connexion v3.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt unit test with connexion v3

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: move connexion v3 dependency to hatch_build

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test view dataset.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
fix:refactor the code for testing

Satoshi-Sh committed:
Created two app with differnt middleware settings

sudiptob2 committed:
fix: adapt redirection tests with starlette tet client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: does not green the test but fix the attribute error.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test units with connextion v3 test client.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
fixed test_dag_endpoint.py

Satoshi-Sh committed:
fix:test_dag_run_endpoint.py

Satoshi-Sh committed:
Fixed test_dag_source_endpoint.py

Satoshi-Sh committed:
Put back validate_responses=True.

potiuk committed:
Fix session handling in test_session_inaccessible_after_logout

There were two problem was with session handling:

* the get_session_cookie - did not get the right cookie - it returned
  "session" string. The right fix was to change cookie_jar into
  cookie.jar because this is where apparently TestClient of starlette
  is holding the cookies (visible when you debug)

* The client does not accept "set_cookie" method - it accepts passing
  cookies via "cookies" dictionary - this is the usual httpx client
  - see  https://www.starlette.io/testclient/ - so we have to set
  cookie directly in the get method to try it out

I added few more calls to show what's going on and to see that the
call works before logout

The other two tests shoudl be fixed similarly.

potiuk committed:
Fix error handling for new connection 3 approach

Error handling for Connexion 3 integration needed to be reworked.
It's likely not a final version - but the way it behaves is much
the same as it works in main:

* for API errors - we get application/problem+json responses
* for UI erros - we have rendered views
* for redirection - we have correct location header (it's been
  missing)
* the api error handled was not added as available middleware
  in the www tests

It should fix all test_views_base.py tests which were failing
on lack of location header for redirection.

potiuk committed:
Add "flask_client_with_login" for tests that neeed flask client

Some tests require functionality not available to Starlette test
client as they use Flask test client specific features - for those
we have an option to get flask test client instead of starlette
one.

potiuk committed:
Fix wrong response is tests_view_cluster_activity

The problem in the test was that Starlette Test Client opens a new
connection and start new session, while flask test client
uses the same database session. The test did not show data because
the data was not committed and session was not closed - which also
failed sqlite local tests with "database is locked" error.

This solution can be applied to other tests where data is differnet
than expected and there is a missing commit / close when data is
prepared.

potiuk committed:
Fix partially test_extra_links

The tests were failing again because the dagrun created was not
committed and session not closed. This worked with flask client
that used the same session accidentally but did not work with
test client from Starlette. Also it caused "database locked"
in sqlite / local tests.

There are still two tests failing with different response - to be
investigated.

potiuk committed:
Fix test_views_grid

Another cases where sessions were not closed

potiuk committed:
Fix test views dataaset

Another session (implicit) not committed.

potiuk committed:
Fix views_custom_user_views tests

The problem in those tests was that the check in security manager
was based on the assumption that the security manager was shared
between the client and test flask application - because they
were coming from the same flask app. But when we use starlette,
the call goes to a new process started and the user is deleted in
the database - so the shortcut of checking the security manager
did not work.

The change is that we are now checking if the user is deleted by
calling /users/show (we need a new users READ permission for that)
 - this way we go to the database and check if the user was indeed
deleted.

potiuk committed:
Partially fix test_views_log.py

This PR partially fixes sessions and request parameter for
test_views_log. Some tests are still failing but for different reasons -
to be investigated.

potiuk committed:
Switch to non-deprecated auth manager

potiuk committed:
Fix error handling for api connexion test

Example fixes to fix tests testing error handling

sudiptob2 committed:
fix: assert response text properly.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
Reverted errro messages

sudiptob2 committed:
fix: adapt test in test_views_home.py

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
Fixed test_event_log_endpoint.py

Satoshi-Sh committed:
fix test_extra_link_endpoint.py

sudiptob2 committed:
fix: adapt test in test_views_log.py

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
Fixed test_health_endpoint.py

Satoshi-Sh committed:
Fixed test_pool_endpoint.py except for one case

Satoshi-Sh committed:
Reverted error message

Satoshi-Sh committed:
Replaced json with json().

Satoshi-Sh committed:
Fixed test_version_endpoint.py and test_xcom_endpoint.py

Satoshi-Sh committed:
Revereted error message and added session.close()

Satoshi-Sh committed:
Fixed test_rpc_api_endpoint.py

sudiptob2 committed:
fix: adapt test_view_tasks.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test_view_tasks.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

potiuk committed:
Fix most test_task_instance_endpoint tests

There were two reasons for the test failed:

* when the Job was added to task instance, the task instance was not
  merged in session, which means that commit did not store the added
  Job

* some of the tests were expecting a call with specific session
  and they failed because session was different. Replacing the
  session with mock.ANY tells pytest that this parameter can be
  anything - we will have different session when when the call will
  be made with ASGI/Starlette

Satoshi-Sh committed:
Fixed response validator errors

potiuk committed:
Fix parameter validation

* added default value for limit parameter across the board. Connexion
  3 does not like if the parameter had no default and we had not
    provided one - even if our custom decorated was adding it. Adding
  default value and updating our decorator to treat None as `default`
  fixed a number of problems where limits were not passed

* swapped openapi specification for /datasets/{uri} and /dataset/events.
  Since `{uri}` was defined first, connection matched `events` with
  `{uri}` and chose parameter definitions from `{uri}` not events

* few other smaller fixes

potiuk committed:
Fix most test_log_enpoint tests

The problem here was that some sessions should be committed/closed but
also in order to run it standalone we wanted to create log templates
in the database - as it relied implcitly on log templates created by
other tests.

Also handling of the response without conteent type had to be fixed.

Remaining issue is 401 vs 403 forbidden returned. To be looked at later.

potiuk committed:
Fix missing session.close() / commit() in mapped instance endpoint

Satoshi-Sh committed:
Fixed test_variable_endpoint.py

potiuk committed:
Fix all FAB provider tests

* Using flask client helped.

Satoshi-Sh committed:
Fixed test_dag_run_schema.py

potiuk committed:
Fix most session problems

Satoshi-Sh committed:
Fixed test_rpc_api_endpoint.py

Satoshi-Sh committed:
Fixed two failing tests

sudiptob2 committed:
fix: adapt url encoded assertions.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

potiuk committed:
Fix integration test

sudiptob2 committed:
fix: adapt test views variables.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

potiuk committed:
Fix static checks

sudiptob2 committed:
fix: used admin_flask_client to fix the filing tests.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

potiuk committed:
Fix more integration tests

sudiptob2 committed:
fix: adapt test_views_task_norun.py

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test_views_rendered.py

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt test_views_robots.py

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
Fix test_views_rate_limit.py

potiuk committed:
Better fix for rate_view_one

The fix checks for the 429 HTTP exception that should be returned
in this case.

This also reverts commit a9aa27d99da3724749a10a7f94c96b6d91eae6ef.

potiuk committed:
Fix test_views_dagrun, test_views_tasks and test_views_log

Fixed by switching to use flask client for testing rather than
starlette. Starlette client in this case has some side effects that are
also impacting Sqlite's session being created in a different
thread and deleted with close_all_sessions fixture.

potiuk committed:
Fix test_views_dagrun

Fixed by switching to use flask client for testing rather than
starlette. Starlette client in this case has some side effects that are
also impacting Sqlite's session being created in a different
thread and deleted with close_all_sessions fixture.

potiuk committed:
Fix test_process_form_invalid_extra_removed

Fixed by switching the test to flask_admin_client. Removes
sqlalchemy session creted from a different thread.

potiuk committed:
Fix test_views_paused

Switching to flask client rather than starlette, helped to fix the
issue.

sudiptob2 committed:
fix: error in tests due to latest merge.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: adapt new tests after latest merge.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

Satoshi-Sh committed:
Fixed minor issues

sudiptob2 committed:
fix: adapt new tests after latest merge.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

potiuk committed:
Fix integration tests with kerberos/trino

It was enough to use flask test  client

Satoshi-Sh committed:
Fixed test_with_default_role
